### PR TITLE
support environment variable for incremental.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @hellosnow @ansyral @vicancy @qinezh @superyyrrzz @herohua

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 1]
```diff
@@ -59,23 +59,24 @@ public class DocumentBuilder : IDisposable
...
-            {
-                var expanded = Environment.ExpandEnvironmentVariables(intermediateFolder);
-                if (expanded.Length == 0)
-                {
-                    expanded = ".";
-                }
-                _intermediateFolder = Path.GetFullPath(expanded);
-            }
+            _intermediateFolder = intermediateFolder;
+            //if (intermediateFolder != null)
```
Comment (herohua) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: why keep this?

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 2]
```diff
@@ -477,13 +477,18 @@ public void SaveIntermediateModel(IncrementalBuildContext incrementalContext)
                         }
 
                         // use copy rather than move because if the build failed, the intermediate files of last successful build shouldn't be corrupted.
-                        File.Copy(Path.Combine(incrementalContext.LastBaseDir, lfn), Path.Combine(incrementalContext.BaseDir, fileName));
+                        File.Copy(
+                            Path.Combine(Environment.ExpandEnvironmentVariables(incrementalContext.LastBaseDir), lfn),
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: >LastBaseDir [](start = 99, length = 11)

can expand in incrementalContext initialization?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: removed.

---
In reply to: [102641811](https://github.com/dotnet/docfx/pull/1339#discussion_r102641811) [](ancestors = 102641811)

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs (hunk 2 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: UT will be fail.

---
In reply to: [102642132](https://github.com/dotnet/docfx/pull/1339#discussion_r102642132) [](ancestors = 102642132)

**Comment by ansyral** [comment]
:shipit:

**Pull request merged**